midi_app_controller/models/app_state.py

```diff
@@ -19,6 +19,8 @@ class AppState(YamlBaseModel):
         Selected MIDI output port.
     recent_binds_for_controller: dict[Path, Optional[Path]]
         Mapping between each controller and which binds setup was last used.
+    recent_midi_ports_for_controller: Optional[dict[Path, dict[str, str]]]
```

Comment:
`recent_binds_for_controller` is not `Optional` so I guess this also shouldn't be.